File: Common/AppStatusLib/src/main/java/com/yc/appstatuslib/broadcast/WifiBroadcastReceiver.java
package com.yc.appstatuslib.broadcast;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.yc.appstatuslib.AppStatusManager;

public class WifiBroadcastReceiver extends BroadcastReceiver {

    private final AppStatusManager mManager;

    public WifiBroadcastReceiver(AppStatusManager mManager) {
        this.mManager = mManager;
    }

    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals("android.net.wifi.WIFI_STATE_CHANGED")) {
            switch(intent.getIntExtra("wifi_state", 4)) {
                case 0:
                case 2:
                case 4:
                default:
                    break;
                case 1:
                    this.notifyGpsSwitchState(false);
                    break;
                case 3:
                    this.notifyGpsSwitchState(true);
            }
        }
    }

    private void notifyGpsSwitchState(boolean state) {
        if (this.mManager != null) {
            this.mManager.dispatcherWifiState(state);
        }
    }
}

File: CommonLib/AppStatusLib/src/main/java/com/yc/appstatuslib/broadcast/WifiBroadcastReceiver.kt
package com.yc.appstatuslib.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager
import com.yc.appstatuslib.AppStatusManager

/**
 * @author: yangchong
 * email  : yangchong211@163.com
 * time   : 2017/5/18
 * desc   : 网络监听广播
 * revise :
 */
class WifiBroadcastReceiver(private val mManager: AppStatusManager?) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == WifiManager.WIFI_STATE_CHANGED_ACTION) {
            val intExtra = intent.getIntExtra(WifiManager.EXTRA_WIFI_STATE, 4)
            when (intExtra) {
                0, 2, 4 -> {

                }
                1 -> notifyWifiSwitchState(false)
                3 -> notifyWifiSwitchState(true)
                else -> {

                }
            }
        }
    }

    private fun notifyWifiSwitchState(state: Boolean) {
        mManager?.dispatcherWifiState(state)
    }
}